include/odp/api/spec/traffic_mngr.h

```diff
@@ -625,6 +625,34 @@ odp_tm_t odp_tm_find(const char            *name,
  */
 int odp_tm_capability(odp_tm_t odp_tm, odp_tm_capabilities_t *capabilities);
 
+/** Start a TM system.
+ *
+ * odp_tm_start() needs to be used to start an already created / found TM system
+ * and in stopped state. By default platforms can choose to be already in
+ * started state with immediately after created. Calling this function
+ * on an already started TM system will be ignored.
+ *
+ * @param odp_tm  The odp_tm_t value of the TM system to be started.
+ * @return        0 upon success, < 0 upon failure.
+ */
+int odp_tm_start(odp_tm_t odp_tm);
+
+/** Stop a TM system.
+ *
+ * odp_tm_stop() can to be used to stop a TM system that is already started.
+ * Stopping a TM system will cause a TM system to stop / pause processing
+ * packets. This can be used to idle a TM system. Applications cannot call
+ * tm enqueue on a TM system which is in stopped state.
```

Comment:
I wonder if it is a bit too strict a requirement to applications to not call tm_enqueue() in the stopped state. AFAICS, calling packet output is ok even when the pktio is in the stopped state. Perhaps (?) this could just say that packets sent in the stopped state may get dropped or may stay in queues until the TM goes to the started state. If we do disallow tm_enqueue() in stopped state, should we ban also odp_ipsec_out_inline() when TM pipelining could happen?

Reply:
I think at least in our HW, the behavior should be same as odp_pktio_stop(). PKTIO stop description says that new packets may not be accepted as well.

Reply:
I, with an application writer hat on, would think that the "may not be accepted" language in pktio means that I am allowed to call packet output functions but the packets may not go anywhere (they get dropped or I get an error). To say that tm_enqueue() cannot be called at all (or maybe the whole thing would crash?) would be a stronger requirement and might force an application writer do do additional synchronization between threads when a TM needs to be stopped to make sure that tm_enqueue() is never called when not allowed.

Reply:
Ack, I get what you mean. I'll make the text same as PKTIO ?

Reply:
Note that TM output may happen also indirectly, through inline IPsec output. I think it would be good to specifiy that when TM is in the stopped state, packets sent to the TM (directly or indirectly) may get silently dropped or may get transmitted without TM treatment or the sending operation may fail.